sdk/messaging/azeventgrid/autorest.md

```diff
@@ -20,34 +18,110 @@ use: "@autorest/go@4.0.0-preview.52"
 version: "^3.0.0"
 slice-elements-byval: true
 remove-non-reference-schema: true
+```
+
+Make sure the content type is setup properly for publishing single and multiple events.
+
+```yaml
 directive:
-  # we have to write a little wrapper code for this so we'll hide the public function
-  # for now.
-  - from: client.go
+  - from: 
+    - client.go
     where: $
-    transform: return $.replace(/PublishCloudEvents\(/g, "internalPublishCloudEvents(");
-  - from: swagger-document
+    transform: | 
+      return $.replace(
+        /(func \(client \*Client\) publishCloudEventsCreateRequest.+?)return req, nil/s, 
+        '$1\nreq.Raw().Header.Set("Content-type", "application/cloudevents-batch+json; charset=utf-8")\nreturn req, nil');
```

Comment:
Do you know if this will be added to the swagger at some point?

Reply:
It's on my TODO list. I need to file some issues - there were a few more that seemed like "do I need to patch this long term?".

Reply:
Filed as https://github.com/Azure/azure-sdk-for-go/issues/22007